nixos/modules/system/boot/systemd/initrd.nix

```
@@ -398,6 +419,9 @@ in {
           ManagerEnvironment=${lib.concatStringsSep " " (lib.mapAttrsToList (n: v: "${n}=${lib.escapeShellArg v}") cfg.managerEnvironment)}
         '';
 
+        # Make the system closure available as an environment variable.
+        "/etc/systemd/system-environment-generators/nixos-environment-generator".source = "${envGenerator}/bin/nixos-environment-generator";
```

```suggestion
        "/etc/systemd/system-environment-generators/nixos-environment-generator".source = lib.getExe envGenerator;
```
TIL that meta.mainProgram is prefilled for writeShellScriptBin